### ISSUE TITLE
Fix usage of `+IGNORE_WARNING` directive with the `..doctest-remote-data ::` directive

### ISSUE THREAD

**Issue by tinumide** [astropy/pytest-doctestplus#147]
@saimn @pllim - we're kind of stuck here. This code chunck is skipped as expected with the new directive, but the test comes back as "passed" rather than "skipped". Have you recall any similar situation, or the reason for this?

_Originally posted by @bsipocz in https://github.com/astropy/pytest-doctestplus/pull/137#discussion_r551495732_

**Comment by tinumide** [comment]
Testing the new remote data directive with the +IGNORE_WARNINGS directive, the test should be skipped if remote data is not requested. It is skipped but the assertion outcome is passed instead of skipped.

**Comment by saimn** [comment]
Can you explain more in detail the issue ? It's not clear to me from the other PR.

**Comment by pllim** [comment]
tl;dr -- A test that should have been skipped is shown as passed even when it is skipped, which is very misleading. Only happens when the embedded code has `+IGNORE_WARNINGS` directive. If you run this with #137 , you can see the behavior. Run it with and without `+IGNORE_WARNING` to see the difference.

Content of `testme.rst`:
```rst
#This test is skipped when remote data is not requested, as it should
#but when remote data is requested the +FLOAT_CMP directive fails

.. doctest-remote-data::

    >>> x = 1/3.
    >>> x    # doctest: +FLOAT_CMP +IGNORE_WARNING
    0.333333
    #
    # do some failing to check whether the whole block is indeed skipped
    >>> 1 + 1
    3

```

* `pytest testme.rst --doctest-plus --doctest-rst --remote-data` fails as expected.
* `pytest testme.rst --doctest-plus --doctest-rst` with `+IGNORE_WARNINGS` shows it as passing (green) -- WRONG
* `pytest testme.rst --doctest-plus --doctest-rst` without `+IGNORE_WARNINGS` shows it as skipped (yellow) -- CORRECT
